api/v1alpha5/virtualmachine_hardware_types.go

```diff
@@ -36,6 +75,26 @@ const (
 	SCSIControllerTypeLsiLogicSAS     SCSIControllerType = "LsiLogicSAS"
 )
 
+// MaxSlots returns the maximum number of devices per SCSI controller type.
+// Note: The controller itself occupies one slot (unit number 7).
+//
+// PVSCSI supports 64 devices per controller starting vSphere 6.7+.
```

Comment:
Please use the name of the type, i.e. 

```golang
The ParaVirtualSCSI type supports 63 devices.
```

* You do no need to go into vSphere versions at all, so drop that and the next line.
* You should also drop the distinction between 64/63. Just say it is 63.